### DISCUSSION TITLE
Looking for help on filters for pnp search

### DISCUSSION THREAD

**Discussion by bigbear48858** [microsoft-search/pnp-modern-search#3300] [Q&A]
I have a people directory created and working on filters. I am pretty new at this, and not finding much information. I want filters for the following;

1. Office Location
2. Teams Channels (Show Members of Channel)

Additionally would like a separate search result webpart to display today's birthdays, and work anniversaries (I know birthdays will probably need to be pulled from a custom field in AD)

**Comment by kasperbolarsen** [comment]
People:SPS-Location is mapped to BaseOfficeLocation on my tenant, and I can't remember having made that mapping, so I think it is the default.
BaseOfficeLocation is refinable out of the box, so that should be easy enough.

However, getting a list of Teams Channels is another story. That kind of info is not stored in the User Profile Application, and I doubt even Graph can provide an answer.

It could be possible to get the birthday and work anniversaries from the UPA so is you could provide the PR I think we would welcome it as a scenario.

**Reply by bigbear48858** [reply]
I've been trying with the BaseOfficeLocation, it shows in Managed Properties, but doesn't seem to show in the pnp search options and just typing it in for the filter configuration gives no results. As for the teams channels, I can from microsoft graph able to query and get results for the teams im a member of using https://graph.microsoft.com/v1.0/me/joinedTeams although I have no clue how to take that information and define it for a filter in pnp.

**Reply by kasperbolarsen** [reply]
Remember that any property must be in the selected properties list in the results web part, in order to be used in the refiners.

**Reply by bigbear48858** [reply]
That's the issue I am having with the BaseOfficeLocation as its not showing as an option.

![image](https://github.com/microsoft-search/pnp-modern-search/assets/135710111/d6b114e1-e5f2-4183-a29f-734729d41fc5)

**Reply by bigbear48858** [reply]
it is shown in managed properties though
![image](https://github.com/microsoft-search/pnp-modern-search/assets/135710111/21a3a1d5-1448-45f0-be06-5bf33f592cce)

**Reply by kasperbolarsen** [reply]
Try just typing it in the selected properties list

**Reply by bigbear48858** [reply]
typing it in let me add it, but still produces no results with the filter webpart. and yes users have data in the feild
![image](https://github.com/microsoft-search/pnp-modern-search/assets/135710111/434548d7-b8af-4510-9c50-09b0ab2da0a2)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/135710111/399b3859-0d20-480f-baa1-63421325e2f6)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/135710111/eecc34c7-ae6b-4260-9f26-4e1f8d1ef104)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/135710111/3c301b7c-912b-4267-8e9e-85d5f5f72401)

**Reply by kasperbolarsen** [reply]
Does any other refiner work? Otherwise you might have forgotten to connect the refiner wp to the result wp, AND the result wp to the refiner wp?

**Reply by bigbear48858** [reply]
Yes, it is connected. I can filter by job description, department, first name, or last name. cant get anything for a location to work though

**Reply by bigbear48858** [reply]
I think I may have it worked out, BaseOfficeLocation was not a refined value. I mapped People:Office to RefinableString00 and forced a site crawl,. looks like I am starting to get some people to show up now will see if I get everyone.

**Reply by kasperbolarsen** [reply]
So it was the base assumption on my part, that I had not mapped BaseOfficeLocation myself that was wrong 🤔 
Sorry

**Comment by kasperbolarsen** [comment]
Regarding the birthday request I recommend looking at this instead,  https://github.com/pnp/sp-dev-fx-webparts/blob/main/samples/react-birthdays/README.md

or perhaps this: https://ms365thinking.blogspot.com/2023/10/using-pnp-modern-search-to-display.html    :-)

**Comment by bigbear48858** [comment]
I found this was very inconsistent at best. Only a handful of records came through. I also discovered that in a Hybrid environment AD OnPrem the "Office" field gets synced to "Office Location" in Azure AD. But in Sharepoint that field syncs to "Office" or "OfficeNumber." Even with the field populated PNP is displaying no data, I assume the field may be defined as a numeric value despite contacting a string. The "Office Location" (mapped to BaseOfficeLocation) in the user's Sharepoint Profile is not synced from Azure AD at all, leaving the value blank. It wasn't easy but I was able to populate this field by using a powerapps flow. I was able to update all my users by pulling a list of users from a Dynamic 365 group containing all my OnPrem synced users, creating an array, and then feeding the data back into a send to HTTP request to SharePoint as seen below
![image](https://github.com/microsoft-search/pnp-modern-search/assets/135710111/414440eb-8e21-49d1-b845-de01f2d32e6f)

**Reply by kasperbolarsen** [reply]
SharePoint User Profile Sync and custom AD property jobs are a couple of huge topics and have been a pain in the behind for as long as I have been in the business.
However, as far as search goes, and this project in particular, we expect the User Profile data to be in place and updated :-)

**Comment by kasperbolarsen** [comment]
And regarding work anniversaries: https://ms365thinking.blogspot.com/2023/10/using-pnp-modern-search-to-display_27.html
#SharingIsCaring
